src/components/UI/Header/Header.tsx

```diff
@@ -1,18 +1,80 @@
 import css from './Header.module.css'
 import { ISeverity } from './../../../model/severity'
-import { getSeverityImage, getResearchImg } from './../../../utils/utils'
+import { getSeverityImage } from './../../../utils/utils'
+import {
+	IAnalysisStep,
+	IDependencyPage,
+	IEosPage,
+	IIaCPage,
+	ISecretsPage,
+	PageType
+} from '../../../model'
+import cveCss from '../Summary/CveVulnerability.module.css'
+import { ReactComponent as JfrogResearchIcon } from '../../../assets/icons/jfrog_research_icon.svg'
+import { ReactComponent as ApplicableIcon } from '../../../assets/icons/applicable.svg'
+import { ReactComponent as NotApplicableIcon } from '../../../assets/icons/not_applicable.svg'
+
 export interface Props {
 	text: string
-	Severity: ISeverity
-	isResearch: boolean
+	pageData: Partial<IDependencyPage | ISecretsPage | IIaCPage | IEosPage>
 }
 
+const LocationSpan = (props: { location: IAnalysisStep }): JSX.Element => (
+	<span className={css.locationLabel}>
+		<b>Location:</b> {props.location.file}
+	</span>
+)
+
+const CveInformation = (props: { data: IDependencyPage }): JSX.Element => (
+	<div className={css.cveInformationContainer}>
+		<span className={css.cveLabel}>Component: {props.data.component}</span>
+		{props.data.version && <span className={css.cveLabel}>Version: {props.data.version}</span>}
+		<span className={css.cveLabel}>
+			CVSS: &nbsp;
+			{props.data.cve?.cvssV3Score ? (
+				<div className={cveCss.redBox}>{props.data.cve.cvssV3Score} (v3)</div>
+			) : (
+				<div className={cveCss.redBox}>{props.data.cve?.cvssV2Score} (v2)</div>
+			)}
+		</span>
+		{props.data.extendedInformation?.jfrogResearchSeverity && (
+			<span className={css.cveLabel}>
+				JFrog severity rank:{' '}
+				{getSeverityImage(props.data.extendedInformation.jfrogResearchSeverity, true)}
+			</span>
+		)}
+	</div>
+)
+
 export default function Header(props: Props): JSX.Element {
+	const showLocationInPages = [PageType.IaC, PageType.Secrets, PageType.Eos].includes(
+		props.pageData.pageType as PageType
+	)
+	const showCVEInfoInPages = PageType.Dependency === props.pageData.pageType
+	const showJFrogResearchIcon =
+		(props.pageData as IDependencyPage).extendedInformation !== undefined
+	const applicableData = (props.pageData as IDependencyPage).cve?.applicableData
```

Comment:
The idea of on Header component is to miminize code duplication, I am not sure that header for each page type is the correct approach as it will result in code duplication

Reply:
We often fall into a trap. A trap that hinges on their fear of duplication.
Duplication is generally a bad thing in software. We don’t like duplicated code.

Consider the dependency header as an example. It follows a slightly different route to generate the research icon. But the EOS header has nothing to do with it. EOS needs a different thing to view, RuleID. Return to it in a few months, and you’ll find that they are very different from each other.

If two apparently duplicated sections of code evolve along different paths, if they change at different rates, and for different reasons, then they are not true duplicates.

Is that true duplication? Most likely it is accidental.
Merging all the header logic in one place introduces two significant issues:

1. We lose the page type, and we have to use 'as' on line 72 with `(props.pageData as IEosPage)`. This 'as' operator bypasses compiler checks, which is considered risky. However, it's worth noting that we had previously determined the page type in the flow, in App.tsx, but unfortunately, this information has been lost.

2. We break esLint by adding ignore on line 71. esLint tells us that we are doing unnecessary conditional, value is always true.

At this point, it should raise a red flag, indicating that we're making a mistake.

Reply:
The header component used to serve all of the pages in the past as well. I kept the same style you used before... It reduces duplication because all headers has some shared styles, html and logic